torch/utils/_config_module.py

```diff
@@ -13,6 +13,32 @@
 from typing_extensions import deprecated
 from unittest import mock
 
+from torch._utils_internal import justknobs_check
+
+
+@dataclass
+class Config:
+    """Represents a config with richer behaviour than just a default value.
+
+    i.e.
+    foo = Config(justknob="//foo:bar", default=False)
+    install_config_module(...)
+
+    This configs must be installed with install_config_module to be used
+
+    Arguments:
+        justknob: the name of the feature / JK. In OSS this is unused.
+        default: is the value to default this knob to in OSS.
+
+    The semantics of this knob, are if no one has manually set it, it will resolve once to the underlying JK value.
+    It will not change the JK value at runtime. If the knob has been manually set, it will ignore JK and use the manually set value.
```

Comment:
btw this diverges from our current behavior

Reply:
I'm happy to make it not sticky :) (My poor memory, is that in the original discussion you wanted it sticky. I have a mild preference for it not being  sticky).

Or do you mean, overrides now apply over justknob changes? (which I don't think we've ever hard consistencty, so it depends on the integration site - i.e. some require both config + JK, some just check JK).

Reply:
Let's make it not sticky for now and talk about it more. Although I used to think that not sticky would lead to more predictable behavior, talking a lot with James March it seems that there's an overriding assumption that JKs will apply to jobs as their running. Since jobs can be pretty long running having a way to get them out of angry path without having to kill them could be useful? IDK, I definitely think revisiting this is good. For now, keep old behavior for ease of rollout.

Reply:
Yeah being able to fix - long running critical jobs, is a very useful property here :).

Reply:
Done.